mesonbuild/build.py

```diff
@@ -227,8 +228,22 @@ def __init__(self, environment: environment.Environment):
         self.stdlibs = PerMachine({}, {})
         self.test_setups: T.Dict[str, TestSetup] = {}
         self.test_setup_default_name = None
-        self.find_overrides = {}
-        self.searched_programs = set() # The list of all programs that have been searched for.
+
+        # All overwritten programs, per-machine
+        find_overrides: PerMachineDefaultable[T.Dict[str, T.Union['ExternalProgram', 'Executable', 'File']]] = PerMachineDefaultable()
```

Comment:
Could we take this opportunity to rename it to `program_overrides` because `find_overrides` means nothing, especially since we also have `dependency_overrides`.